src/datasets/arrow_dataset.py

```diff
@@ -2848,6 +2848,26 @@ def add_elasticsearch_index(
             )
         return self
 
+    def add_item(self, item: dict):
+        """Add item to Dataset.
+
+        .. versionadded:: 1.6
+
+        Args:
+            item (dict): Item data to be added.
+
+        Returns:
+            :class:`Dataset`
+        """
+        item_table = InMemoryTable.from_pydict({k: [v] for k, v in item.items()})
+        # Cast item
+        type = self.features.type
+        schema = pa.schema({col_name: type[col_name].type for col_name in self._data.column_names})
```

Comment:
```suggestion
        schema = pa.schema(self.features.type)
```
It's maybe simpler to do it this way ?

Reply:
Sure!!! 😅